src/mds/QuiesceDbManager.cc

```diff
@@ -52,11 +52,22 @@ static QuiesceTimeInterval time_distance(QuiesceTimePoint lhs, QuiesceTimePoint
   }
 }
 
-bool QuiesceDbManager::db_thread_has_work() const
+bool QuiesceDbManager::db_thread_has_work(bool leader_bootstrapping) const
 {
+  if (leader_bootstrapping) {
+    // if the leader is bootstrapping, `db_thread_should_exit` cannot be true
+    ceph_assert(!db_thread_should_exit);
+    if ((agent_callback.has_value() && agent_callback->if_newer < db.version())
+    || (cluster_membership.has_value() && cluster_membership->epoch != membership.epoch)) {
+      return true;
+    }
+    // if (pending_db_updates.size() > 0 || pending_acks.size() > 0 || pending_requests.size() > 0)
+    // if the the requests/acks aren't empty then skipping a sleep will cause
+    // a busy-loop of the quiesce manager thread.
+    return false;
+  }
+
```

Comment:
the reason behind the removal of the acks/requests checks - if we're here, it asserts that there is no bootstrap delay i.e. there the pending acks/reqs should be empty. Maybe it is feasible to assert them before the return statement?